src/red.ts

```diff
@@ -213,9 +213,13 @@ export const serializedRedEnvSourceText = (function redEnvFactory(blueEnv: Membr
     }
 
     function copyRedOwnDescriptors(shadowTarget: RedShadowTarget, blueDescriptors: PropertyDescriptorMap) {
-        for (const key in blueDescriptors) {
+        const keys = ownKeys(blueDescriptors);
```

Comment:
this one here is fine... we don't care about the proto chain, because it is a controlled mechanism, we read the descriptor map, and we call this method right after, no chain involved.